src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -822,6 +837,13 @@ public static MongoClientSettings FromUrl(MongoUrl url)
 #pragma warning restore CS0618 // Type or member is obsolete
                     resolveHosts = connectionMode == ConnectionMode.Direct || connectionMode == ConnectionMode.Standalone;
                 }
+
```

Comment:
Why not just change line 838 to:

    resolveHosts = connectionMode == ConnectionMode.Direct || connectionMode == ConnectionMode.Standalone || url.LoadBalanced;